docs/architecture/design-proposals/dashboard/deployment-improvements.md

```diff
@@ -0,0 +1,197 @@
+        "user": { "type": "string" },
+        "password": { "type": "string" },
+        "database": { "type": "string" },
+        "port": { "type": "integer" }
+      }
+    },
+    "persistence": {
+      "type": "object",
+      "properties": {
+        "size": {
+          "type": "string",
+          "title": "Disk Size",
+              "anyOf": [
+            {"pattern": ".*"},
+            {"enum": ["10Gi", "50Gi", "100Gi"]},
+          ]
+        }
+      }
+    },
+    "resources": {
+      "type": "object",
+      "title": "Required Resources",
+      "description": "Deployment resource requests",
+      "properties": {
+        "requests": {
+          "type": "object",
+          "properties": {
+            "memory": {
+              "type": "string",
+              "anyOf": [
+                { "pattern": ".*" },
+                { "enum": ["128Mi", "512Mi", "1Gi"] }
+              ]
+            },
+            "cpu": {
+              "type": "string",
+              "anyOf": [
+                { "pattern": ".*" },
+                { "enum": ["100m", "300m", "500m"] }
+              ]
+            }
+          }
+        },
+      }
+    },
+    "enableMetrics": { "type": "boolean", "title": "Enable Metrics" },
+    "replicas": { "type": "integer", "title": "Number of Replicas"}
+  },
+
+  "required": ["wordpressUsername", "wordpressEmail", "persistence", "resources"],
+
+  "kubeapps": {
+    "username": "wordpressUsername",
+    "password": "wordpressPassword",
+    "email": "wordpressEmail",
+    "enableExternalDatabase": "mariadb.enabled",
+    "externalDatabase": "externalDatabase",
+    "disk": "persistence.size",
+    "resources": "resources",
+    "enableMetrics": "enableMetrics",
+    "replicas": "replicas"
+  }
+}
+```
+
+This `values.schema.json` is a valid JSON schema for the `values.yaml` of WordPress. It can be more complete since it's possible to include every single parameter of the values.yaml file.
+
+In order to be able to provide a richer presentation of the form we are including the key `kubeapps`. This allows us to know which parameters we need to represent in the form and how to map them to valid values. In case there is a parameter that is not known we should be able to parse it as a generic input.
+
+## Deployment Workflow
+
+This section defines how the deployment logic will work based on the JSON Schema definition above.
+
+1. The deployment component loads the details of the Chart. If the chart doesn't contain the file `values.schema.json`, continue as today; Only the advanced deployment is allowed. In other case, go to the next step.
+2. Load the `values.schema.json`. If it doesn't contain the `kubeapps` key, auto-generate the form using the tool `react-jsonschema-form`. From this point, two different tabs will be shown to the user. If the user changes tab from `basic` to `advanced` the new form will be shown but the values won't be ported. If the `kubeapps` key is defined go to the next step.
```

Comment:
As above, I'd base this on both the `values.schema.json` being present, as well as the (separate) definition of which fields are most relevant for a simple deploy.